ScrollOfResurrection.lua

```diff
@@ -0,0 +1,741 @@
+            if summonPlayer == nil then
+                chatHandler:SendSysMessage("Target not found. Check spelling and capitalization.")
+                SOR_cleanup()
+                return false
+            end
+
+            if SOR_recruiterAccount[summonPlayer:GetAccountId()] ~= player:GetAccountId() then
+                chatHandler:SendSysMessage("The requested player is not resurrected by you.")
+                SOR_cleanup()
+                return false
+            end
+
+            if SOR_timeStamp[summonPlayer:GetAccountId()] == 0 or SOR_timeStamp[summonPlayer:GetAccountId()] == 1 then
+                chatHandler:SendSysMessage("The requested player is not resurrected by you anymore.")
+                SOR_cleanup()
+                return false
+            end
+
+            -- do the zone/combat checks and possibly summon
+            local mapId = player:GetMapId()
+            -- allow to proceed if the player is on one of the maps listed above
+            if SOR_hasValue(Config_maps, mapId) then
+                --allow to proceed if the player is not in combat
+                if not player:IsInCombat() then
+                    local group = player:GetGroup()
+                    if group == nil then
+                        chatHandler:SendSysMessage("You must be in a party or raid to summon your resurrected friend.")
+                        return false
+                    end
+                    local groupPlayers = group:GetMembers()
+                    for _, v in pairs(groupPlayers) do
+                        if v:GetName() == commandArray[3] then
+                            if player:GetPlayerIP() == v:GetPlayerIP() and Config.checkSameIp == 1 and SOR_timeStamp[summonPlayer:GetAccountId()] >= 2 then
+                                chatHandler:SendSysMessage("Possible abuse detected. Aborting. This action is logged.")
+                                if SOR_sameIpCounter[summonPlayer:GetAccountId()] == nil then
+                                    SOR_sameIpCounter[summonPlayer:GetAccountId()] = 1
+                                    CharDBExecute('UPDATE `'..Config.customDbName..'`.`scroll_of_resurrection_links` SET ip_abuse_counter = '..SOR_sameIpCounter[summonPlayer:GetAccountId()]..' WHERE `account_id` = '..summonPlayer:GetAccountId()..';')
+                                    SOR_cleanup()
+                                    return false
+                                else
+                                    SOR_sameIpCounter[summonPlayer:GetAccountId()] = SOR_sameIpCounter[summonPlayer:GetAccountId()] + 1
+                                    CharDBExecute('UPDATE `'..Config.customDbName..'`.`scroll_of_resurrection_links` SET ip_abuse_counter = '..SOR_sameIpCounter[summonPlayer:GetAccountId()]..' WHERE `account_id` = '..summonPlayer:GetAccountId()..';')
+                                    SOR_cleanup()
+                                    return false
```

Comment:
here you could put the `return` after the if and else, same for `SOR_cleanup()`, because you execute them for both cases
```suggestion
                                if SOR_sameIpCounter[summonPlayer:GetAccountId()] == nil then
                                    SOR_sameIpCounter[summonPlayer:GetAccountId()] = 1
                                    CharDBExecute('UPDATE `'..Config.customDbName..'`.`scroll_of_resurrection_links` SET ip_abuse_counter = '..SOR_sameIpCounter[summonPlayer:GetAccountId()]..' WHERE `account_id` = '..summonPlayer:GetAccountId()..';')
                                else
                                    SOR_sameIpCounter[summonPlayer:GetAccountId()] = SOR_sameIpCounter[summonPlayer:GetAccountId()] + 1
                                    CharDBExecute('UPDATE `'..Config.customDbName..'`.`scroll_of_resurrection_links` SET ip_abuse_counter = '..SOR_sameIpCounter[summonPlayer:GetAccountId()]..' WHERE `account_id` = '..summonPlayer:GetAccountId()..';')
                                
                                SOR_cleanup()
                                return false
```

Reply:
Good catch!